conans/test/integration/toolchains/gnu/test_makedeps.py

```diff
@@ -0,0 +1,428 @@
+    client = TestClient()
+    client.save({"conanfile.py": conanfile})
+    client.run("create .")
+    client.run("install --requires=mylib/0.1@ -g MakeDeps")
+
+    makefile_path = os.path.join(client.current_folder, CONAN_MAKEFILE_FILENAME)
+    assert os.path.isfile(makefile_path)
+    makefile_content = load(makefile_path)
+    prefix = pathlib.Path.home().drive if platform.system() == "Windows" else ""
+    assert 'CONAN_NAME_MYLIB = mylib' in makefile_content
+    assert 'CONAN_VERSION_MYLIB = 0.1' in makefile_content
+    assert f'CONAN_LIB_DIRS_MYLIB = \\\n\t$(CONAN_LIB_DIR_FLAG){prefix}/my_absoulte_path/fake/mylib/lib \\\n\t$(CONAN_LIB_DIR_FLAG)$(CONAN_ROOT_MYLIB)/lib2' in makefile_content
+    assert f'CONAN_INCLUDE_DIRS_MYLIB = $(CONAN_INCLUDE_DIR_FLAG){prefix}/my_absoulte_path/fake/mylib/include' in makefile_content
+    assert 'CONAN_BIN_DIRS_MYLIB = $(CONAN_BIN_DIR_FLAG)$(CONAN_ROOT_MYLIB)/bin' in makefile_content
+
+    def assert_is_abs(path):
+        assert os.path.isabs(path) is True
+
+    lines = makefile_content.splitlines()
+    for line_no, line in enumerate(lines):
+        includedir_pattern = "CONAN_INCLUDE_DIRS_MYLIB = $(CONAN_INCLUDE_DIRS_FLAG)"
+        if line.startswith(includedir_pattern):
+            assert_is_abs(line[len(includedir_pattern):])
+            assert line.endswith("include")
+        elif line.startswith("\t$(CONAN_LIB_DIRS_FLAG)") and 'my_absoulte_path' in line:
+            assert_is_abs(line[len("\t$(CONAN_LIB_DIRS_FLAG)"):-2])
+            assert line.endswith("lib \\")
+        elif line.startswith("\t$(CONAN_LIB_DIRS_FLAG)") and line.endswith('lib2'):
+            assert "\t$(CONAN_LIB_DIRS_FLAG)$(CONAN_ROOT_MYLIB)/lib2" in line
+
+
+def test_make_empty_dirs():
+    """
+    MakeDeps should support cppinfo empty dirs
+    """
+    conanfile = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+
+        class TestMakeDepsConan(ConanFile):
+            name = "mylib"
+            version = "0.1"
+
+            def package_info(self):
+                self.cpp_info.includedirs = []
+                self.cpp_info.libdirs = []
+                self.cpp_info.bindirs = []
+                self.cpp_info.libs = []
+                self.cpp_info.frameworkdirs = []
+        """)
+    client = TestClient()
+    client.save({"conanfile.py": conanfile})
+    client.run("create .")
+    client.run("install --requires=mylib/0.1@ -g MakeDeps")
+
+    makefile_path = os.path.join(client.current_folder, CONAN_MAKEFILE_FILENAME)
+    assert os.path.exists(makefile_path) is True
+    makefile_content = load(makefile_path)
+    assert 'CONAN_ROOT_MYLIB' in makefile_content
+    assert 'SYSROOT' not in makefile_content
+    assert 'CONAN_INCLUDE_DIRS' not in makefile_content
+    assert 'CONAN_LIB_DIRS' not in makefile_content
+    assert 'CONAN_BIN_DIRS' not in makefile_content
+    assert 'CONAN_LIBS' not in makefile_content
+    assert 'CONAN_FRAMEWORK_DIRS' not in makefile_content
```

Comment:
Just treats as empty, for instance:

    $(CC) $(TORTILLA) main.c -o main
    
In case `TORTILLA` is not declared, but still, `make` will ignore it and use empty as replacement.